src/classes/CountryStateDistrict.cls

```diff
@@ -0,0 +1,115 @@
+public with sharing class CountryStateDistrict {
+    //dynamiclly field Maps
+    Map<string,Map<string,List<string>>> mapOfStatewithCountry=new Map<string,Map<string,List<string>>>();
+
+    Map<string,List<string>> mapOfDistrictsWithState=new Map<string,List<string>>();
+
+    List<string> ListOfDistricts=new List<string>();
+    public  CountryStateDistrict(string ch,integer length) {
+        List<State__c> details=[SELECT Name ,(SELECT Name FROM Districts__r WHERE Name LIKE :ch+ '%'),
+                                Country_Name__r.Name FROM State__c WHERE Name LIKE :ch+'%' and Country_Name__r.Name LIKE :ch+'%'] ; 
+        for(State__c states: details){
+            for(District__c district: states.Districts__r){
+                ListOfDistricts.add(district.Name);
+            }
+            mapOfDistrictsWithState.put(states.Name,ListOfDistricts);
+            mapOfStatewithCountry.put(states.Country_Name__r.Name,mapOfDistrictsWithState);
+        }
+        //Displaying country ,state and district combination where start with s
+        system.debug('Start with'+ch+'all countries,state and districts'+mapOfStatewithCountry);
+        //Find such a districts which length is more or equal to some number
+        List<District__c> districts=[SELECT Name FROM District__c];
+        for(District__c district:  districts){
+            if(length<=district.Name.length() ){
+                system.debug(district.Name);
+            }
+        }
+    }
+    // Calculate Total number of States And Districts using Aggregate Queries
+    List<Sobject> sobjectListForDml=new List<Sobject> ();
+    public void totalNoOfStateAndDistrict(){
```

Comment:
No requirement of this method. This should be done from the JsonGenerator class only.